Kitodo/src/test/java/org/kitodo/production/services/file/FileServiceIT.java

```diff
@@ -56,4 +63,42 @@ public void testRenameFileWithLockedFile() throws IOException {
         }
 
     }
+
+    @Test
+    public void testCreateDirectories() throws IOException {
+        Path testBaseDirectoryPath = Paths.get(ConfigCore.getParameter(ParameterCore.DIR_XML_CONFIG));
+
+        // delete existing directories so that directories are actually created
+        Path firstDirectoryOfTestPath = testBaseDirectoryPath.resolve("several");
+        if (Files.isDirectory(firstDirectoryOfTestPath)) {
+            TreeDeleter.deltree(firstDirectoryOfTestPath);
+        }
+
+        String testBaseDirectoryURI = testBaseDirectoryPath.toUri().toString();
+        String createDirectories = "several/directories/can/be/created/at/once";
+        new FileService().createDirectories(URI.create(testBaseDirectoryURI + createDirectories));
+        assertTrue(Files.isDirectory(firstDirectoryOfTestPath.resolve("directories/can/be/created/at/once")));
+
+        // clean up
+        TreeDeleter.deltree(firstDirectoryOfTestPath);
+    }
+
+    @Test
+    public void testCreateDirectoriesWithTrailingSlash() throws IOException {
+        Path testBaseDirectoryPath = Paths.get(ConfigCore.getParameter(ParameterCore.DIR_XML_CONFIG));
```

Comment:
The mapping to the kitodo directory is happening on the module

Reply:
Thats why your test isn't failing with indexOutOfBounds.
In the core we don't know how the file system is managed, only the FileManagement knows.

Reply:
I did not understand it yet. What must the test look like to (currently) fail?

Reply:
She meant that all magic happens in File Management module, not here. That's why this test will not fail for whatever reason. Simply saying, there is not way to do it.

Reply:
The question is, if the method I am testing is allegedly flawed, then it must be possible to write a test that results in the error. If not, the method must be correct. If the error occurs in the file management module, then there is the same thing, then you would have to be able to write a test for it, which now produces an error first. The thing is: the method works for me and does what it should. But Kathrin said she had written a "tiny little" test that immediately threw an exception. If so, I want to know what the test looks like so that I can reproduce the error and then repair the method.